alias support for ORCA optional with GUC parameter

dkovalev1 opened the pull request:
Add support for table aliases in ORCA

Currently ORCA generated query plan does not show aliases, unlike original
PostgreSQL planner. Usually it is much more descriptive to have the alias in the
plan, which are specified by user in the query and might have meaning.

This patch adds table aliases support to ORCA. Table aliases are added to both
internal structures and DXL schema and for debug representation of internal
objects. Since explain command uses RangeTableEntry fields of PlannedStmt to
extract the alias, we also fill rte->alias field during PlannedStmt creation.

Table aliases are widely used in the tests, so simple adding them requires lots
of tests changes and patch becomes rather huge. To eliminate this, GUC parameter
optimizer_enable_table_alias has been introduced. Default value is on, it
enables table aliases. Setting it to off makes aliases not used, as before the
patch.

The optimizer_enable_table_alias parameter intended mainly for automated tests,
so tests have to set it explicitly for all the cases. This is done in two steps:
1. When optimizer=on is used, i.e. we are running ORCA-enabled tests,
the OPTIMIZER_ENABLE_TABLE_ALIAS environment variable is set to off. 
2. At the demo cluster startup, in demo_cluster.sh, if
OPTIMIZER_ENABLE_TABLE_ALIAS is set, the line optimizer_enable_table_alias=off
is appended to the file clusterConfigPostgresAddonsFile which is used to
distribute configuration across demo cluster. Basically,
optimizer_enable_table_alias is needed only for the master node and could be set
on per-connection basis, but this approach is simple, consistent and has minimum
required changes.

dkovalev1 committed:
alias support for ORCA optional with GUC parameter

BenderArenadata commented:
Failed job Build for x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1075177

RekGRpth commented:
<h2>Problems with Symbols, Medium Severity <span class="failed"> 1 </span></h2><hr>
<span class="h_name">guc_tables.h</span>, <span class="lib_name">postgres</span><br>
<span class="section" onclick="javascript:showContent(this, 'c_1')">
<span class="ext">[−]</span> ConfigureNamesBool_gp <span class="attr">[data]</span> <span class="failed"> 1 </span></span>
<br>
<div id="c_1" style="display: block; visibility: visible;">


  | Change | Effect
-- | -- | --
1 | Size of this global data has been changed from 38152 bytes to 38304 bytes. | Applications will obtain a different value and execution may change.

</div>

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64716

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1075189

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1075806

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1075190

BenderArenadata commented:
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1075182

RekGRpth commented:
Why did you decide to open a new pull request, while there were still unanswered questions in the old one https://github.com/arenadata/gpdb/pull/682?

dkovalev1 committed:
setup ignore symbols for a baseline

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64834

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081251

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081252

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081243

BenderArenadata commented:
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081244

dkovalev1 committed:
update expected output for test

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64837

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081371

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081369

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081362

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081372

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081365

BenderArenadata commented:
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1081363

RekGRpth commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +329,15 @@ CREATE TABLE t1 (a int);
+
+
+
+create table foo_alias (a int, b int);
```
You should first use `drop table if exist` in `begin_ignore` and `end_ignore` block

RekGRpth commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +329,15 @@ CREATE TABLE t1 (a int);
+
+set optimizer_enable_table_alias=on;
+explain delete from foo_alias bbb using foo_alias aaa  where aaa.a=bbb.a;
+
```
You should reset the GUC here.
```suggestion
reset optimizer_enable_table_alias;
```

dkovalev1 replied:
Good point!

RekGRpth commented on src/test/regress/expected/gp_explain.out:
```diff
@@ -529,3 +529,35 @@ EXPLAIN ANALYZE INSERT INTO t1 SELECT 1/gp_segment_id
+               ->  Seq Scan on foo_alias  (cost=0.00..431.00 rows=4 width=18)
+               ->  Hash  (cost=431.00..431.00 rows=4 width=4)
+                     ->  Seq Scan on foo_alias foo_alias_1  (cost=0.00..431.00 rows=4 width=4)
+ Optimizer: Pivotal Optimizer (GPORCA)
```
You should put to this file output for postgres legacy planner, not ORCA!

dkovalev1 committed:
replace template for non-optimized test run and add optional drop table

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64848

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1082197

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1082198

dkovalev1 committed:
reset parameter optimizer_enable_table_alias before and after the test to make the test independent

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64942

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1085750

dkovalev1 commented:
> Why did you decide to open a new pull request, while there were still unanswered questions in the old one #682?
#682 had to make a lot of changes to tests and total number of changed files is 50+ This PR is to introduce parameter to optionally enable alias support and keep number of changed files smaller. 
It would be possible to change old one PR, but idea is to keep changes to tests for the case if we decide not to use the parameter.

dkovalev1 committed:
Update baseline for ABI compare

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64952

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1086534

RekGRpth commented on src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:
```diff
@@ -4724,6 +4728,16 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
+		Store the alias information only when table alias is used to allow 
+		EXPLAIN find it
+	*/
+	if (optimizer_enable_table_alias && NULL != md_alias)
```
Why is this check needed?! The md_alias pointer is dereferenced above, which means it is definitely not NULL here!

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1086535

RekGRpth commented on src/backend/gporca/libnaucrates/src/operators/CDXLOperatorFactory.cpp:
```diff
@@ -1483,7 +1483,21 @@ CDXLOperatorFactory::MakeDXLTableDescr(CDXLMemoryManager *dxl_memory_manager,
+	CDXLTableDescr *ret =
+		GPOS_NEW(mp) CDXLTableDescr(mp, mdid, mdname, user_id);
+
+	const XMLCh *table_alias_xml =
```
```suggestion
	const XMLCh *xml_str_alias_name =
		ExtractAttrValue(attrs, EdxltokenAlias, EdxltokenTableDescr, true);
```

RekGRpth replied:
I suggest not only rename

dkovalev1 replied:
ExtractAttrValue seems to be doing absolutely the same in this case, but with another one stack frame. What is the benefit of calling it?

RekGRpth replied:
To be closer to similar code above.

dkovalev1 replied:
There is a code above:
`
	const XMLCh *execute_as_user_xml =
		attrs.getValue(CDXLTokens::XmlstrToken(EdxltokenExecuteAsUser));
`
which is optional, like alias and unlike name. This code is simple, does not use unnecessary entities (ExtractAttrValue, EdxltokenTableDescr, true)  and follows OO paradigm in xerces/Orca.
Not sure why ever to use procedural wrapper in OO code.

RekGRpth commented on src/backend/gporca/libgpopt/include/gpopt/metadata/CTableDescriptor.h:
```diff
@@ -138,6 +141,19 @@ class CTableDescriptor : public CRefCount,
+
+	void
+	SetAlias(const CWStringConst *alias_str)
+	{
```
```suggestion
	{
		GPOS_ASSERT(NULL == m_alias);
```

RekGRpth commented on src/backend/gporca/libgpopt/src/metadata/CTableDescriptor.cpp:
```diff
@@ -77,6 +78,12 @@ CTableDescriptor::CTableDescriptor(
+	if (NULL != m_alias)
+	{
+		GPOS_DELETE(m_alias);
+		m_alias = NULL;
```
Why do we need it?

dkovalev1 replied:
May be I'm not completely understand a question, but if the field is not null in the destructor, we delete it's memory. And since null is a valid value for this pointer,  regardless of whether it will be used in other cleanup methods we put null to it's value. ATM it might be not used, but leaving invalid dangling pointer seems worse then making one assignment.

RekGRpth commented on src/backend/gporca/libnaucrates/src/operators/CDXLTableDescr.cpp:
```diff
@@ -86,6 +87,26 @@ CDXLTableDescr::MdName() const
+{
+	if (NULL != m_mdalias)
+	{
+		GPOS_DELETE(m_mdalias);
```
instead delete I suggest ensure it is null
```c
GPOS_ASSERT(NULL == m_mdalias);
```

dkovalev1 replied:
With deleting it's possible to overwrite existing value. Not sure if it's used anywhere now, but assert doesn't seem to be better idea here, and also it works only for the debug version.

RekGRpth replied:
Are we going to re-record something? Is this even a valid case?

dkovalev1 replied:
I don't know if it valid or not, but why to prohibit while it costs nothing to support?

RekGRpth replied:
> why to prohibit while it costs nothing to support?

Then you should provide a test for this branch of code

dkovalev1 replied:
Am I understand correctly that tests and asserts can be considered as mutual exclusive concepts for trivial changes?

RekGRpth replied:
Something like that.

dkovalev1 replied:
Okay although I am not quite happy with it

RekGRpth commented on src/backend/gpopt/translate/CTranslatorUtils.cpp:
```diff
@@ -136,6 +136,14 @@ CTranslatorUtils::GetTableDescr(CMemoryPool *mp, CMDAccessor *md_accessor,
 
+	if (NULL != rte->alias)
+	{
+		CMDName *table_alias =
```
I suggest rename it to `table_mdalias`

dkovalev1 replied:
not sure if it makes big sense for local variable but ok

RekGRpth commented on src/backend/gporca/libnaucrates/src/operators/CDXLOperatorFactory.cpp:
```diff
@@ -1483,7 +1483,21 @@ CDXLOperatorFactory::MakeDXLTableDescr(CDXLMemoryManager *dxl_memory_manager,
+
+	if (NULL != table_alias_xml)
+	{
+		CMDName *md_alias = CDXLUtils::CreateMDNameFromXMLChar(
```
```suggestion
		CMDName *mdalias = CDXLUtils::CreateMDNameFromXMLChar(
```

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1086532

dkovalev1 committed:
implement suggestions from review

RekGRpth commented on src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:
```diff
@@ -4724,6 +4728,16 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
+	*/
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		rte->alias = alias;
```
Why is rte->eref equal to rte->alias in this case?

dkovalev1 replied:
Sorry, could not understand the question. The line with rte->eref assignment exists for 2 years and was not affected by this PR. EXPLAIN uses rte->alias to print alias information, that's why I put it here.

RekGRpth replied:
I meant why do you change rte->eref? Maybe I shouldn't have done this?

dkovalev1 replied:
Still don't understand, I can't see where rte->eref is changed in this commit.

RekGRpth replied:
`rte->eref = alias;` and you changed `alias->aliasname`

RekGRpth replied:
so question: why do you change rte->eref while you are using rte->alias?

dkovalev1 replied:
In the parsenodes.h we can see comment:

>  *	  eref is the table reference name and column reference names (either
>  *	  real or aliases).

And since we are to support aliases in Orca it seems logical to see actual alias in the eref.

RekGRpth replied:
Then why don't you use rte->eref below?

dkovalev1 replied:
Sorry, again not getting you. Where should I use rte->eref and why? The point of this PR is to fill rte->alias with actual alias name in AS clause.

RekGRpth replied:
> Where should I use rte->eref and why?

https://github.com/arenadata/gpdb/pull/780/files#diff-c66cca54786d2aba98ba3a9b644ce710180f227f699b3247f302d19a498dc69eR142

RekGRpth replied:
Why are you changing `rte->eref->aliasname`? Maybe it’s enough to just change `rte->alias->aliasname`?

dkovalev1 replied:
As I wrote rte->eref is a table reference name either real or alias. And now alias is supported and available nearby. I found no reason to use real name still. Regression shown that it is valid approach.

RekGRpth replied:
I get it: are you using `rte->alias` below as `NULL != table_descr->MdAlias()` condition?

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/64956

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1086801

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1086802

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1086793

dkovalev1 committed:
prohibit updating alias in CDXLTableDescr, make an alias copy in translator

RekGRpth commented on src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:
```diff
@@ -4672,8 +4672,12 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
+
 	alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
-		table_descr->MdName()->GetMDName()->GetBuffer());
+		md_alias->GetMDName()->GetBuffer());
```
How about such optimization?
```c
if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
{
rte->alias = alias;
alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
		table_descr->MdAlias()->GetMDName()->GetBuffer());
}
else
{
alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
		table_descr->MdName()->GetMDName()->GetBuffer());
}
```

dkovalev1 replied:
Interesting... Is it better?

RekGRpth replied:
Your code sets `alias->aliasname` by `table_descr->MdAlias()` independetly of `optimizer_enable_table_alias`

dkovalev1 replied:
Setting it dependently now

RekGRpth replied:
I think, my solution is more simple and less invasive. Another solution:
```c
const CMDName *md_alias = table_descr->MdName();
if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
{
md_alias = table_descr->MdAlias();
rte->alias = alias;
}
alias->aliasname = CTranslatorUtils::CreateMultiByteCharStringFromWCString(
		md_alias->GetMDName()->GetBuffer());
```

dkovalev1 replied:
okay, removed ternary operator

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65210

BenderArenadata commented:
Failed job ORCA code linter on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1096677

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1096676

dkovalev1 committed:
make clang-format happy

dkovalev1 committed:
set alias->aliasname depending of optimizer_enable_table_alias

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65305

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1101119

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1101120

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1101117

bimboterminator1 commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +329,21 @@ CREATE TABLE t1 (a int);
 FROM gp_dist_random('gp_id');
 DROP TABLE t1;
+
+
```
Write the comments describing the test.

bimboterminator1 replied:
the most correct variant is to add comments to sql file as well. And format your comments properly: --  table... - why 2 spaces?

bimboterminator1 commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +329,21 @@ CREATE TABLE t1 (a int);
+explain delete from foo_alias bbb using foo_alias aaa  where aaa.a=bbb.a;
+
+drop table foo_alias;
+reset optimizer_enable_table_alias;
```
add LF

bimboterminator1 replied:
still not fixed

bimboterminator1 commented on .abi-check/6.26.0_arenadata54/postgres.symbols.ignore:
```diff
@@ -0,0 +1 @@
+ConfigureNamesBool_gp
```
why do we add 2 files, not 1?

dkovalev1 replied:
During the work baseline has changed and I had to add another one file

bimboterminator1 replied:
In other words, there is no way we can avoid creating another file?

dkovalev1 replied:
One file per baseline. May be we can safely remove previous one now.

RekGRpth replied:
> May be we can safely remove previous one now

I think so

dkovalev1 replied:
Done

bimboterminator1 commented on src/backend/utils/misc/guc_gp.c:
```diff
@@ -2974,6 +2976,16 @@ struct config_bool ConfigureNamesBool_gp[] =
+		{"optimizer_enable_table_alias", PGC_USERSET, DEVELOPER_OPTIONS,
+			gettext_noop("Enable using table aliases to make plan explain more descriptive"),
+			NULL
+		},
```
Why don't you mention GUC properties like `GUC_NOT_IN_SAMPLE` as it's done for others?

bimboterminator1 commented on src/backend/gporca/libnaucrates/src/operators/CDXLTableDescr.cpp:
```diff
@@ -198,6 +215,12 @@ CDXLTableDescr::SerializeToDXL(CXMLSerializer *xml_serializer) const
 	xml_serializer->AddAttribute(CDXLTokens::GetDXLTokenStr(EdxltokenTableName),
 								 m_mdname->GetMDName());
 
+	if (NULL != m_mdalias)
```
In 7X aliases are used in place of table names. Such behaviour affects all the serialization procedures, including not only DXL representation, but also various operator trees, memo contents, etc.
For example:
```
explain (costs off) select sum(t2.a), t1.a, t1.b, t1.c from funcdep1 t1 join funcdep2 t2 on t1.b = t2.b group by t1.a;
Algebrized preprocessed query:
+--CLogicalGbAgg( Global ) Grp Cols: ["a" (0), "b" (1), "c" (2)][Global], Minimal Grp Cols: [], Generates Duplicates :[ 0 ] 
   |--CLogicalNAryJoin
   |  |--CLogicalGet "t1" ("t1"), Columns: ["a" (0), "b" (1), "c" (2), "d" (3), "ctid" (4), "xmin" (5), "cmin" (6), "xmax" (7), "cmax" (8), "tableoid" (9), "gp_segment_id" (10)] Key sets: {[0], [4,10]}
   |  |--CLogicalGet "t2" ("t2"),
....
....
Or CMemo contents:
Group 1 (#GExprs: 2):
  0: CLogicalGet "t2" ("t2"), Columns: ["a" (11), "b" (12), "c" (13), "d" (14), "ctid" (15), "xmin" (16), "cmin" (17), "xmax" (18), "cmax" (19), "tableoid" (20), "gp_segment_id" (21)] Key sets: {[4,10]} [ ]
  1: CPhysicalTableScan "t2" ("t2")
```
The question we need to answer: Do we need such behaviour for all the serializable objects, or putting aliases only to DXL is enough for us? We need to clarify that moment.

dkovalev1 committed:
Add GUC options to the new optimizer_enable_table_alias parameter

dkovalev1 committed:
Add comments to test

dkovalev1 committed:
replace non-pretty ternary operator

RekGRpth commented on src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:
```diff
@@ -4672,8 +4672,14 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
 	// get table alias
+	const CMDName *md_alias = table_descr->MdName();
+		
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias()) {
```
```suggestion
	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
	{
```

RekGRpth commented on src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:
```diff
@@ -4724,6 +4730,16 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
+	*/
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		rte->alias = (Alias *) copyObject(alias);
```
How about to move it to above same condition? And without copying

dkovalev1 replied:
Prefer to make a deep copy after filling fields than before. Copy is used here because in the similar places where both eref and alias initialized copy is used.

RekGRpth replied:
Why do we need deep copy here?

dkovalev1 replied:
Same reason as for:
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/analyze.c#L1095
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1225
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1678
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1761
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/src/backend/parser/parse_relation.c#L1850
Perhaps it's done to allow modify and use them independently in the code.

RekGRpth replied:
Everything mentioned is C code, how does this apply to our case? Where are they changed and used independently in ORCA?

dkovalev1 replied:
RLE here is prepared for EXPLAIN which is in backend/command and is  not part of ORCA. It's not used in ORCA.

RekGRpth replied:
> It's not used in ORCA

https://github.com/arenadata/gpdb/pull/780/files#diff-c66cca54786d2aba98ba3a9b644ce710180f227f699b3247f302d19a498dc69eR139-R145

dkovalev1 replied:
rte is used not only in the ORCA, but also in C-code at least in ExplainNode. I have feeling that it's much more safe to have a deep copy, similar to other places in C code.

dkovalev1 committed:
remove obsolete file for baseline check

dkovalev1 committed:
Fix formatting

dkovalev1 committed:
fix comments

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65390

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1104044

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1104051

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1106058

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1106057

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1106055

dkovalev1 committed:
Change default value for optimizer_enable_table_alias to on

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65476

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1108534

RekGRpth commented:
I think, new GUC should be documented

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1108526

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1108528

BenderArenadata commented:
Failed job Regression tests with Postgres on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1108527

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1108532

dkovalev1 committed:
Fix for test expected template

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65516

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1110230

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1110231

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1110224

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1110225

dkovalev1 committed:
Set optimizer_enable_table_alias in scripts instead of configuration

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65560

BenderArenadata commented:
Failed job Behave tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1111222

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1111223

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1111224

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1111217

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1111218

dkovalev1 committed:
move OPTIMIZER_ENABLE_TABLE_ALIAS var to proper place

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65602

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1113056

BenderArenadata commented:
Failed job Behave tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1113055

dkovalev1 commented:
> @dkovalev1, Update the patch description and describe shortly the changes have been made in the patch, including the CI related changes.
Updated.

dkovalev1 commented:
> I think, new GUC should be documented

This GUC was intended for tests only, to reduce changes to tests. Have you an idea who and why in real world going to use it?

RekGRpth commented:
> > I think, new GUC should be documented
> 
> This GUC was intended for tests only, to reduce changes to tests. Have you an idea who and why in real world going to use it?

Then maybe rename this GUC to optimizer_disable_table_alias?

dkovalev1 committed:
Documentation update

BenderArenadata commented:
Allure report https://allure-ee.adsw.io/launch/65692

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1116806

BenderArenadata commented:
Failed job Behave tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1116805

bimboterminator1 commented on src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:
```diff
@@ -4724,6 +4731,16 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
+	*/
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		rte->alias = (Alias *) copyObject(alias);
```
Why both `rte->eref` and `rte->alias` are filled with the same value? And why we are making copy for `rte->alias`, but not for `rte->eref`?

RekGRpth replied:
> Why both `rte->eref` and `rte->alias` are filled with the same value?

https://github.com/arenadata/gpdb/pull/780#discussion_r1495261708

RekGRpth replied:
> And why we are making copy for `rte->alias`, but not for `rte->eref`?

https://github.com/arenadata/gpdb/pull/780#discussion_r1502552841

bimboterminator1 replied:
Yep, my bad. I.e. eref and alias are meant to have the same value?

dkovalev1 replied:
Yes, in this particular case they have equal value. But the fields has different purpose and potentially different life cycle, that's why value is copied.

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1116807

bimboterminator1 commented:
@dkovalev1, i've made several edits to the PR description, take a look please

RekGRpth commented:
> Default value is on, which works as before this patch

I do not understand this

RekGRpth commented:
>confiruration

configuration

RekGRpth commented on gpdb-doc/dita/ref_guide/config_params/guc-list.xml:
```diff
@@ -471,6 +471,9 @@
             <li><xref href="#optimizer_enable_orderedagg" type="section"
                 >optimizer_enable_orderedagg</xref></li>
+            <li>
+              <xref href="#optimizer_enable_table_alias" format="dita"
```
What is mean `format="dita"`?

dkovalev1 replied:
It seems to be yet another one documentation format. 
https://github.com/arenadata/gpdb/blob/2fd27bbc76ae4547cd99d1edac260805f6655cd6/gpdb-doc/README.md?plain=1#L5
I've just copied it from another topic in the same document.

RekGRpth replied:
why not `type="section"` as above and below?

dkovalev1 replied:
because if was copy-pasted in other place, where "dita" was above and below and then placed alphabetically

dkovalev1 commented:
> > Default value is on, which works as before this patch
> 
> I do not understand this

Fixed

dkovalev1 commented:
> > confiruration
> 
> configuration

Fixed

RekGRpth commented:
> except for debug representation of some
> internal objects (e.x. debug print of the group content)

Why did you decide not to add aliases to these internal objects as well? I think this will be useful when debugging.

RekGRpth commented:
> This parameter optimizer_enable_table_alias intended mainly for automated tests,
> so tests have to set it explicitly for all the cases.

I didn't quite understand this.

dkovalev1 committed:
Use alias in CLogicalGet to enable alias in debug print of the plan

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66419

BenderArenadata commented:
Failed job ORCA code linter on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1149019

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1149017

dkovalev1 committed:
clang-format change

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66428

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1149446

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1149447

RekGRpth commented:
> Table aliases are added to both
> internal structures and DXL schema, except for debug representation of some
> internal objects (e.x. debug print of the group content).

Actualize PR description.

RekGRpth commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -350,5 +356,15 @@ explain delete from foo_alias bbb using foo_alias aaa  where aaa.a=bbb.a;
+-- start_ignore
 drop table foo_alias;
-reset optimizer_enable_table_alias;
+set optimizer_print_plan to default;
```
Why not reset?

dkovalev1 replied:
Set is used more often in tests and is is easier to grep because 'to default;' is more unique.

RekGRpth replied:
Still, you changed it to reset!

RekGRpth commented on src/backend/gporca/libgpopt/src/translate/CTranslatorDXLToExpr.cpp:
```diff
@@ -574,8 +574,13 @@ CTranslatorDXLToExpr::PexprLogicalGet(const CDXLNode *dxlnode)
-	CWStringConst strAlias(m_mp,
-						   table_descr->MdName()->GetMDName()->GetBuffer());
+	const gpmd::CMDName *ptabalias = table_descr->MdName();
+	if (NULL != table_descr->MdAlias())
```
How about turning it off using the GUC?

dkovalev1 replied:
Debug print is human-readable and  is not used in tests usually, besides the case I made specially for it. Reason for the GUC was successful passage of unrelated tests, and since tests are passed, there is no reason to use GUC here.

RekGRpth replied:
I think there will be a discrepancy between expected and actual behavior.

dkovalev1 replied:
Of course it will. It's just not used in the tests.

bimboterminator1 replied:
We don't need to put GUC condition here, because we can launch GPORCA separately from gpdb by running gporca_test utility with minidump, which has alias in table descriptor.

dkovalev1 commented:
> > Table aliases are added to both
> > internal structures and DXL schema, except for debug representation of some
> > internal objects (e.x. debug print of the group content).
> 
> Actualize PR description.

Done.

bimboterminator1 commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +335,36 @@ CREATE TABLE t1 (a int);
+
+
+-- Check optional table alias support. When optimizer_enable_table_alias is off,
+-- table alias is not supported (this is default)
```
what (this is default )means ?

dkovalev1 replied:
This is a default for the tests, to keep previous behavior.

bimboterminator1 replied:
> This is a default for the tests

It's not obvious that this is default only for the tests. You should clarify that in this comments or remove this words completely.

dkovalev1 replied:
Right. Rewrote it.

bimboterminator1 commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +335,36 @@ CREATE TABLE t1 (a int);
+explain delete from foo_alias bbb using foo_alias aaa  where aaa.a=bbb.a;
+
+-- Check that debug print of physical plan also contains alias
+set optimizer_print_plan=on;
```
do we really need that test? I can't find any other regress tests that use debug pring and I see no reason to introduce such test. Moreover alias in debug print is independent from guc. I suggest to remove it, becuase this test forced you to change message level and add extra matchsubs

dkovalev1 replied:
There was a change specially for the debug print, and this is the way how to verify the change. Are there any guideline restricting from use of message level and/or matchsubs?

bimboterminator1 replied:
> Are there any guideline restricting from use of message level and/or matchsubs? 

No, there aren't. But test would look much cleaner without it.

bimboterminator1 commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +335,36 @@ CREATE TABLE t1 (a int);
+set client_min_messages to default;
+-- end_ignore
+
+set optimizer_enable_table_alias to default;
```
What if the order of tests will change, and tests that have been generated without aliases will be executed after gp_explain?

dkovalev1 committed:
Rephrase comments in test

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66525

BenderArenadata commented:
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1154155

andr-sokolov commented on concourse/scripts/common.bash:
```diff
@@ -87,6 +87,11 @@ function make_cluster() {
 	export BLDWRAP_POSTGRES_CONF_ADDONS=${BLDWRAP_POSTGRES_CONF_ADDONS}
 	export STATEMENT_MEM=250MB
+
+    if [[ "$MAKE_TEST_COMMAND" =~ "optimizer=on" ]]; then
```
The indentation here (spaces) differs from the indentation around (tabs). Use tabs

andr-sokolov commented on src/test/regress/sql/gp_explain.sql:
```diff
@@ -329,3 +335,37 @@ CREATE TABLE t1 (a int);
+-- end_ignore
+
+-- Check that with optimizer_enable_table_alias=off plan doesn't have aliases.
+-- When optimizer_enable_table_alias is off, table alias is not supported.
```
may be "table aliases are not supported"?

dkovalev1 committed:
Fix formatting and indentation.

dkovalev1 committed:
Merge branch 'adb-6.x-dev' into ADBDEV-4487-param

dkovalev1 committed:
better test readability

andr-sokolov commented on src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp:
```diff
@@ -4687,8 +4687,15 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
 	alias->colnames = NIL;
 
 	// get table alias
+	const CMDName *md_alias = table_descr->MdName();
+
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		md_alias = table_descr->MdAlias();
+	}
```
md_alias is rewritten here and the previous value is never used in this case, so the MdName method is called in vain. What about ternary operator?
```suggestion
	const CMDName *md_alias = 
	    (optimizer_enable_table_alias && NULL != table_descr->MdAlias()) ?
	    table_descr->MdAlias() :
	    table_descr->MdName();
```

andr-sokolov replied:
or
```
const CMDName *md_alias = table_descr->MdAlias();
if (!optimizer_enable_table_alias || NULL == md_alias)
{
	md_alias = table_descr->MdName();
}
```

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66616

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1157979

dkovalev1 committed:
optimize expression

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66648

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1161084

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1161085

dkovalev1 committed:
Merge branch 'adb-6.x-dev' into ADBDEV-4487-param

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66875

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1174516

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1174517

dkovalev1 committed:
Add new GUC to ignore list to make ABI checker happy again

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66889

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1175345

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1175339

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1175340

dkovalev1 committed:
Remove physical plan from the test. The test results comparer can handle minor descrepancies in plan as a result of EXPLAIN statement, but unfortunately can not handle it for the physical plan.
It would be nice to have such a possibility one day, but now it's out of the scope of current task.

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66930

dkovalev1 committed:
rename baseline to 6.26.0_arenadata55 for ABI checks

BenderArenadata commented:
Allure report https://allure.adsw.io/launch/66931

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1177785

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1177786

BenderArenadata commented:
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1177777

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1177779

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1178481

dkovalev1 commented:
> New flaky [test](https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1178481)?
> 
I seems to be failing test, not just flaky. The question is  - was in introduced by this PR or was brought by the merge. It was appeared after the merge and it doesn't look at the first glance what could bring this new problem.

BenderArenadata commented:
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1187480

BenderArenadata commented:
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1187481

RekGRpth commented:
Another flaky [test](https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1187480)?
<details>
<summary>Another flaky test?</summary>

```diff
--- \/home\/gpadmin\/gpdb_src\/src\/test\/isolation2\/expected\/fts_manual_probe\.out	2024-03-21 07:27:32.664985517 +0000
+++ \/home\/gpadmin\/gpdb_src\/src\/test\/isolation2\/results\/fts_manual_probe\.out	2024-03-21 07:27:32.672985615 +0000
@@ -101,23 +101,70 @@
  Success:                 
 (1 row)
 
-1&: select gp_request_fts_probe_scan();  <waiting ...>
-2&: select gp_request_fts_probe_scan();  <waiting ...>
-3&: select gp_request_fts_probe_scan();  <waiting ...>
+1&: select gp_request_fts_probe_scan();Process Process-2:
+Traceback (most recent call last):
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
+    self.run()
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
+    self._target(*self._args, **self._kwargs)
+  File "./sql_isolation_testcase.py", line 290, in session_process
+    self.mode, pipe, self.dbname, user=self.user, passwd=self.passwd)
+  File "./sql_isolation_testcase.py", line 400, in __init__
+    self.con = self.connectdb(self.dbname)
+  File "./sql_isolation_testcase.py", line 412, in connectdb
+    passwd = given_passwd)
+InternalError: FATAL:  optimizer failed to init (CSyncHashtable.h:110)
+HINT:  Process xxx will wait for gp_debug_linger=120 seconds before termination.
+Note that its locks and other resources will not be released until then.
+
+  <waiting ...>
+FAILED:  poll() gave POLLNVAL or POLLERR
+2&: select gp_request_fts_probe_scan();Process Process-3:
+Traceback (most recent call last):
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
+    self.run()
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
+    self._target(*self._args, **self._kwargs)
+  File "./sql_isolation_testcase.py", line 290, in session_process
+    self.mode, pipe, self.dbname, user=self.user, passwd=self.passwd)
+  File "./sql_isolation_testcase.py", line 400, in __init__
+    self.con = self.connectdb(self.dbname)
+  File "./sql_isolation_testcase.py", line 412, in connectdb
+    passwd = given_passwd)
+InternalError: FATAL:  optimizer failed to init (CSyncHashtable.h:110)
+HINT:  Process xxx will wait for gp_debug_linger=120 seconds before termination.
+Note that its locks and other resources will not be released until then.
+
+  <waiting ...>
+FAILED:  poll() gave POLLNVAL or POLLERR
+3&: select gp_request_fts_probe_scan();Process Process-4:
+Traceback (most recent call last):
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
+    self.run()
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
+    self._target(*self._args, **self._kwargs)
+  File "./sql_isolation_testcase.py", line 290, in session_process
+    self.mode, pipe, self.dbname, user=self.user, passwd=self.passwd)
+  File "./sql_isolation_testcase.py", line 400, in __init__
+    self.con = self.connectdb(self.dbname)
+  File "./sql_isolation_testcase.py", line 412, in connectdb
+    passwd = given_passwd)
+InternalError: FATAL:  optimizer failed to init (CSyncHashtable.h:110)
+HINT:  Process xxx will wait for gp_debug_linger=120 seconds before termination.
+Note that its locks and other resources will not be released until then.
+
+  <waiting ...>
+FAILED:  poll() gave POLLNVAL or POLLERR
 
 -- piggyback: ensure the probe requests are at a known starting location
 select gp_wait_until_triggered_fault('ftsNotify_before', 3, 1) from master();
- gp_wait_until_triggered_fault 
--------------------------------
- Success:                      
-(1 row)
+ERROR:  fault not triggered, fault name:'ftsNotify_before' fault type:'wait_until_triggered'
+DETAIL:  Timed-out as 10 minutes max wait happens until triggered.
 
 -- piggyback: ensure the ftsLoop is triggered only once
 select gp_wait_until_triggered_fault('ftsLoop_after_latch', 1, 1) from master();
- gp_wait_until_triggered_fault 
--------------------------------
- Success:                      
-(1 row)
+ERROR:  fault not triggered, fault name:'ftsLoop_after_latch' fault type:'wait_until_triggered'
+DETAIL:  Timed-out as 10 minutes max wait happens until triggered.
 select gp_inject_fault('ftsLoop_after_latch', 'resume', 1) from master();
  gp_inject_fault 
 -----------------
@@ -126,10 +173,8 @@
 
 -- piggyback: ensure the ftsLoop is at a known starting location
 select gp_wait_until_triggered_fault('ftsLoop_before_probe', 1, 1) from master();
- gp_wait_until_triggered_fault 
--------------------------------
- Success:                      
-(1 row)
+ERROR:  fault not triggered, fault name:'ftsLoop_before_probe' fault type:'wait_until_triggered'
+DETAIL:  Timed-out as 10 minutes max wait happens until triggered.
 select insert_expected_stats('top_of_ftsLoop', 0, 0);
  insert_expected_stats 
 -----------------------
@@ -158,10 +203,8 @@
  Success:        
 (1 row)
 select gp_wait_until_triggered_fault('ftsLoop_after_probe', 1, 1) from master();
- gp_wait_until_triggered_fault 
--------------------------------
- Success:                      
-(1 row)
+ERROR:  fault not triggered, fault name:'ftsLoop_after_probe' fault type:'wait_until_triggered'
+DETAIL:  Timed-out as 10 minutes max wait happens until triggered.
 
 select insert_expected_stats('bottom_of_ftsLoop', 1, 0);
  insert_expected_stats 
@@ -171,7 +214,7 @@
 select * from get_stats;
  seq | seq_name          | expected_start_delta | actual_start_delta | expected_done_delta | actual_done_delta 
 -----+-------------------+----------------------+--------------------+---------------------+-------------------
- 3   | bottom_of_ftsLoop | 1                    | 1                  | 0                   | 0                 
+ 3   | bottom_of_ftsLoop | 1                    | 0                  | 0                   | 0                 
 (1 row)
 
 -- fresh result test: issue a new probe request during the in progress piggyback ftsLoop
@@ -185,7 +228,24 @@
 --------------------------
  Success:                 
 (1 row)
-4&: select gp_request_fts_probe_scan();  <waiting ...>
+4&: select gp_request_fts_probe_scan();Process Process-5:
+Traceback (most recent call last):
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
+    self.run()
+  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
+    self._target(*self._args, **self._kwargs)
+  File "./sql_isolation_testcase.py", line 290, in session_process
+    self.mode, pipe, self.dbname, user=self.user, passwd=self.passwd)
+  File "./sql_isolation_testcase.py", line 400, in __init__
+    self.con = self.connectdb(self.dbname)
+  File "./sql_isolation_testcase.py", line 412, in connectdb
+    passwd = given_passwd)
+InternalError: FATAL:  optimizer failed to init (CSyncHashtable.h:110)
+HINT:  Process xxx will wait for gp_debug_linger=120 seconds before termination.
+Note that its locks and other resources will not be released until then.
+
+  <waiting ...>
+FAILED:  poll() gave POLLNVAL or POLLERR
 
 -- piggyback: resume the suspended piggyback ftsLoop
 select gp_inject_fault('ftsLoop_after_probe', 'resume', 1) from master();
@@ -195,27 +255,16 @@
 (1 row)
 
 1<:  <... completed>
- gp_request_fts_probe_scan 
----------------------------
- t                         
-(1 row)
+FAILED:  Execution failed
 2<:  <... completed>
- gp_request_fts_probe_scan 
----------------------------
- t                         
-(1 row)
+FAILED:  Execution failed
 3<:  <... completed>
- gp_request_fts_probe_scan 
----------------------------
- t                         
-(1 row)
+FAILED:  Execution failed
 
 -- fresh result: ensure the next ftsLoop iteration is at a known starting location
 select gp_wait_until_triggered_fault('ftsLoop_before_probe', 1, 1) from master();
- gp_wait_until_triggered_fault 
--------------------------------
- Success:                      
-(1 row)
+ERROR:  fault not triggered, fault name:'ftsLoop_before_probe' fault type:'wait_until_triggered'
+DETAIL:  Timed-out as 10 minutes max wait happens until triggered.
 
 -- piggyback: query the probe stats before the start of the 'fresh result' ftsLoop
 select insert_expected_stats('piggyback_result', 1, 1);
@@ -226,7 +275,7 @@
 select * from get_stats;
  seq | seq_name         | expected_start_delta | actual_start_delta | expected_done_delta | actual_done_delta 
 -----+------------------+----------------------+--------------------+---------------------+-------------------
- 4   | piggyback_result | 1                    | 1                  | 1                   | 1                 
+ 4   | piggyback_result | 1                    | 0                  | 1                   | 0                 
 (1 row)
 
 -- fresh result: resume the suspended ftsLoop
@@ -237,10 +286,7 @@
 (1 row)
 
 4<:  <... completed>
- gp_request_fts_probe_scan 
----------------------------
- t                         
-(1 row)
+FAILED:  Execution failed
 
 select insert_expected_stats('fresh_result', 2, 2);
  insert_expected_stats 
@@ -250,7 +296,7 @@
 select * from get_stats;
  seq | seq_name     | expected_start_delta | actual_start_delta | expected_done_delta | actual_done_delta 
 -----+--------------+----------------------+--------------------+---------------------+-------------------
- 5   | fresh_result | 2                    | 2                  | 2                   | 2                 
+ 5   | fresh_result | 2                    | 0                  | 2                   | 0                 
 (1 row)
 
 -- show all raw stats for debugging
@@ -269,18 +315,33 @@
 -- reset the internal regular probe interval
 !\retcode gpconfig -r gp_fts_probe_interval;
 GP_IGNORE:-- start_ignore
-GP_IGNORE:20190730:11:15:35:045960 gpconfig:office-5-75:dkrieger-[INFO]:-completed successfully with parameters '-r gp_fts_probe_interval'
+GP_IGNORE:20240321:07:27:32:129038 gpconfig:3b3a9bb25239:gpadmin-[ERROR]:-FATAL:  optimizer failed to init (CSyncHashtable.h:110)
+GP_IGNORE:HINT:  Process 129045 will wait for gp_debug_linger=120 seconds before termination.
+GP_IGNORE:Note that its locks and other resources will not be released until then.
+GP_IGNORE:
+GP_IGNORE:20240321:07:27:32:129038 gpconfig:3b3a9bb25239:gpadmin-[ERROR]:-Failed to connect to database, exiting without action. This script can only be run when the database is up.
+GP_IGNORE:Failed to connect to database, exiting without action. This script can only be run when the database is up.
 GP_IGNORE:
 GP_IGNORE:-- end_ignore
-(exited with code 0)
+(exited with code 1)
 !\retcode gpstop -u;
 GP_IGNORE:-- start_ignore
-GP_IGNORE:20190730:11:15:35:046018 gpstop:office-5-75:dkrieger-[INFO]:-Starting gpstop with args: -u
-GP_IGNORE:20190730:11:15:35:046018 gpstop:office-5-75:dkrieger-[INFO]:-Gathering information and validating the environment...
-GP_IGNORE:20190730:11:15:35:046018 gpstop:office-5-75:dkrieger-[INFO]:-Obtaining Greenplum Master catalog information
-GP_IGNORE:20190730:11:15:35:046018 gpstop:office-5-75:dkrieger-[INFO]:-Obtaining Segment details from master...
-GP_IGNORE:20190730:11:15:35:046018 gpstop:office-5-75:dkrieger-[INFO]:-Greenplum Version: 'postgres (Greenplum Database) 7.0.0-alpha.0+dev.575.g59811832fc build dev'
-GP_IGNORE:20190730:11:15:35:046018 gpstop:office-5-75:dkrieger-[INFO]:-Signalling all postmaster processes to reload
+GP_IGNORE:20240321:07:27:32:129047 gpstop:3b3a9bb25239:gpadmin-[INFO]:-Starting gpstop with args: -u
+GP_IGNORE:20240321:07:27:32:129047 gpstop:3b3a9bb25239:gpadmin-[INFO]:-Gathering information and validating the environment...
+GP_IGNORE:20240321:07:27:32:129047 gpstop:3b3a9bb25239:gpadmin-[INFO]:-Obtaining Greenplum Master catalog information
+GP_IGNORE:20240321:07:27:32:129047 gpstop:3b3a9bb25239:gpadmin-[INFO]:-Obtaining Segment details from master...
+GP_IGNORE:20240321:07:27:32:129047 gpstop:3b3a9bb25239:gpadmin-[CRITICAL]:-gpstop failed. (Reason='FATAL:  optimizer failed to init (CSyncHashtable.h:110)
+GP_IGNORE:HINT:  Process 129056 will wait for gp_debug_linger=120 seconds before termination.
+GP_IGNORE:Note that its locks and other resources will not be released until then.
+GP_IGNORE:') exiting...
 GP_IGNORE:
 GP_IGNORE:-- end_ignore
-(exited with code 0)
+(exited with code 2)
+Traceback (most recent call last):
+  File "./sql_isolation_testcase.py", line 1087, in <module>
+    executor.process_isolation_file(sys.stdin, sys.stdout, options.initfile_prefix)
+  File "./sql_isolation_testcase.py", line 838, in process_isolation_file
+    process.stop()
+  File "./sql_isolation_testcase.py", line 335, in stop
+    self.pipe.send(("", False))
+IOError: [Errno 32] Broken pipe
```
<details>

dkovalev1 committed:
Fix ABI checks

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1193659

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1193660

bimboterminator1 commented on .abi-check/6.26.2_arenadata55/postgres.symbols.ignore:
```diff
@@ -1,2 +1,3 @@
 ConfigureNamesBool_gp
+optimizer_enable_table_alias
```
why do we explicitly include the guc name? Why not `ConfigureNamesBool_gp`? Or this aspect is not important?

Commit:
Merge branch 'adb-6.x-dev' into ADBDEV-4487-param

BenderArenadata commented:
Failed job Build for x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1208306

BenderArenadata commented:
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1208322

BenderArenadata commented:
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1208323
